visualization/app/codeCharta/util/nodeDecorator.ts

```diff
@@ -16,220 +15,210 @@ import { CodeMapHelper } from "./codeMapHelper"
 	public static decorateMap(map: CodeMapNode, metricData: NodeMetricData[], blacklist: BlacklistItem[]) {
-		this.decorateNodesWithIds(map)
-		this.decorateMapWithMissingObjects(map)
-		this.decorateMapWithCompactMiddlePackages(map)
-		this.decorateLeavesWithMissingMetrics(map, metricData)
-		this.decorateMapWithBlacklist(map, blacklist)
-	}
-
-	public static decorateMapWithPathAttribute(file: CCFile) {
-		if (file && file.map) {
-			const root = hierarchy<CodeMapNode>(file.map)
-			root.each(node => {
-				node.data.path =
-					"/" +
-					root
-						.path(node)
-						.map(x => x.data.name)
-						.join("/")
-			})
-		}
-		return file
-	}
+		const root = hierarchy<CodeMapNode>(map)
 
-	private static decorateMapWithBlacklist(map: CodeMapNode, blacklist: BlacklistItem[]) {
 		const flattened = ignore()
 		const excluded = ignore()
 
+		let hasFlattenedPaths = false
+		let hasExcludedPaths = false
+
 		for (const item of blacklist) {
 			const path = CodeMapHelper.transformPath(item.path)
-			item.type === BlacklistType.flatten ? flattened.add(path) : excluded.add(path)
-		}
 
-		hierarchy(map)
-			.descendants()
-			.map(node => {
-				const path = CodeMapHelper.transformPath(node.data.path)
-				node.data.isFlattened = flattened.ignores(path)
-				node.data.isExcluded = excluded.ignores(path)
-			})
-	}
+			if (item.type === BlacklistType.flatten) {
+				hasFlattenedPaths = true
+				flattened.add(path)
+			} else {
+				hasExcludedPaths = true
+				excluded.add(path)
+			}
+		}
 
-	private static decorateNodesWithIds(map: CodeMapNode) {
 		let id = 0
-		hierarchy(map)
-			.descendants()
-			.map(node => {
-				node.data.id = id
-				id++
-			})
-	}
 
-	private static decorateMapWithCompactMiddlePackages(map: CodeMapNode) {
-		const isEmptyMiddlePackage = current => {
-			return (
-				current &&
-				current.children &&
-				current.children.length === 1 &&
-				current.children[0].children &&
-				current.children[0].children.length > 0
-			)
+		const defaultAttributes = { [NodeMetricDataService.UNARY_METRIC]: 1 }
+		const defaultLeafAttributes = { ...defaultAttributes }
+		if (metricData?.length) {
+			for (const { name } of metricData) {
+				defaultLeafAttributes[name] = 0
+			}
 		}
 
-		const rec = current => {
-			if (isEmptyMiddlePackage(current)) {
-				const child = current.children[0]
-				current.children = child.children
-				current.name += "/" + child.name
-				current.path += "/" + child.name
-				if (child.link) {
-					current.link = child.link
-				}
-				current.attributes = child.attributes
-				current.edgeAttributes = child.edgeAttributes
-				current.deltas = child.deltas
-				rec(current)
-			} else if (current && current.children && current.children.length > 1) {
-				for (let i = 0; i < current.children.length; i++) {
-					rec(current.children[i])
+		for (const { data } of root.descendants()) {
+			data.id = id
+			id++
+			if (data.attributes) {
+				data.attributes[NodeMetricDataService.UNARY_METRIC] = 1
```

Comment:
How come? The logic is the same as before and tests will fail by removing this.

Reply:
That's probably because those are unit test and the provided metric data for the test doesn't include the `unary` metric. When we set new metrics, we add the unary metric though. I guess we might have to update the test data, but I gotta check again.

Reply:
Maybe, I would still rather not change anything that requires changes in until tests in this PR.